clang/lib/CIR/Dialect/IR/CIRAttrs.cpp

```diff
@@ -446,6 +446,42 @@ DataMemberAttr::verify(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ItaniumDynamicCastInfoAttr definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult ItaniumDynamicCastInfoAttr::verify(
+    function_ref<InFlightDiagnostic()> emitError,
+    mlir::cir::GlobalViewAttr srcRtti, mlir::cir::GlobalViewAttr destRtti,
+    mlir::FlatSymbolRefAttr runtimeFunc, mlir::FlatSymbolRefAttr badCastFunc,
+    mlir::cir::IntAttr offsetHint) {
+  auto isRttiPtr = [](mlir::Type ty) {
```

Comment:
Is it possible to implement these constraints in tablegen? I know it works for operations but I don't remember whether it works for attributes. `ExceptionInfoPtr` would be a good example to use for comparison.